lib/core/ast_processor_manager/ast_processor_manager.h

```diff
@@ -120,6 +120,30 @@ HandleType makeSpecificSemaXXXHandle(Handler& handler) {
     }};
 };
 
+// TODO: maybe remove dublication with makeSpecificSemaXXXHandle
+template <typename Handler, typename NodeType, typename HandleType>
+HandleType makeDefaultSemaXXXHandle(Handler& handler) {
+    using ExprType = typename std::remove_reference_t<typename func_param_type<Handler, 2>::type>;
+    constexpr size_t n_arguments = func_num_arguments<Handler>::value;
+
+    return HandleType{[&handler, n_arguments](const clang::Attr*,
+                                              const NodeType& node,
+                                              OklSemaCtx& sema,
+                                              SessionStage& stage) -> HandleResult {
+        static_assert(n_arguments == HANDLE_NUM_OF_ARGS, "Handler must have 4 arguments");
+        const auto localNode = clang::dyn_cast_or_null<ExprType>(&node);
+        if (!localNode) {
+            auto baseNodeTypeName = typeid(NodeType).name();
+            auto handleNodeTypeName = typeid(ExprType).name();
+            return tl::make_unexpected(
+                Error{{},
+                      util::fmt("Failed to cast {} to {}", baseNodeTypeName, handleNodeTypeName)
+                          .value()});
+        }
+        return handler(nullptr, *localNode, sema, stage);
```

Comment:
Always passing `nullptr` as `Attr` to the handler.
Please optimize.

Reply:
Removing this argument will break a whole ton of code. I must add new DefaultStmtHandler and DefaultDeclHandler, then update makeDefaultHandler, then update type of default handler map, then AstProcessorManager::runPreActionNodeHandle and runNodeHandler assume that attribute and default handlers have the same interface, which is another ton of fixes.

Reply:
I see. well, it's OK.